components/admin_console/team_channel_settings/update_confirm_modal.jsx

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import {t} from 'utils/i18n.jsx';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
+import ConfirmModal from 'components/confirm_modal.jsx';
+
+import RemoveConfirmModal from './remove_confirm_modal';
+import ConvertConfirmModal from './convert_confirm_modal';
+
+export default class UpdateConfirmModal extends React.PureComponent {
+    static propTypes = {
+
+        /*
+         * Bool whether the modal is shown
+         */
+        show: PropTypes.bool.isRequired,
+
+        /*
+         * Action to call on confirm
+         */
+        onConfirm: PropTypes.func.isRequired,
+
+        /*
+         * Action to call on cancel
+         */
+        onCancel: PropTypes.func.isRequired,
+
+        /*
+         * Channel display name
+         */
+        displayName: PropTypes.string.isRequired,
+
+        /*
+         * Channel public/private state
+         */
+        toPublic: PropTypes.bool.isRequired,
+
+        /*
+         * Indicates if the channel privacy is being updated
+         */
+        isPrivacyChanging: PropTypes.bool.isRequired,
+
+        /*
+         * Number of users to be removed
+         */
+        removeAmount: PropTypes.number.isRequired,
+    }
+
+    render() {
+        const {show, onConfirm, onCancel, displayName, isPrivacyChanging, toPublic, removeAmount} = this.props;
+
+        if (isPrivacyChanging && removeAmount === 0) {
+            return (
+                <ConvertConfirmModal
+                    show={show}
+                    onCancel={onCancel}
+                    onConfirm={onConfirm}
+                    displayName={displayName}
+                    isPublic={!toPublic}
+                />
+            );
+        }
+
+        if (!isPrivacyChanging && removeAmount > 0) {
+            return (
+                <RemoveConfirmModal
+                    amount={removeAmount}
+                    inChannel={true}
+                    show={show}
+                    onCancel={onCancel}
+                    onConfirm={onConfirm}
+                />
+            );
+        }
+
+        const newType = toPublic ? 'public' : 'private';
```

Comment:
Maybe we can move everything here to a new "ConvertAndDremoveConfirmModal" to make it more homogeneous. you are mixing here 2 different levels of abstraction, and is less obvious than everything in the same level of abstraction.